Doppler.HtmlEditorApi/Storage.DapperProvider/DapperCampaignContentRepository.cs

```diff
@@ -65,19 +65,39 @@ public async Task<ContentData> GetCampaignModel(string accountName, int campaign
             editorType: queryResult.EditorType);
     }
 
-    public async Task SaveCampaignContent(string accountName, ContentData contentRow)
+    public async Task<CampaignState> GetCampaignState(string accountName, int campaignId)
     {
-        var campaignStatus = await _dbContext.ExecuteAsync(new FirstOrDefaultCampaignStatusDbQuery(
+        var campaignStateData = await _dbContext.ExecuteAsync(new FirstOrDefaultCampaignStatusDbQuery(
             AccountName: accountName,
-            IdCampaign: contentRow.campaignId
+            IdCampaign: campaignId
         ));
 
-        // TODO: consider returning 404 NotFound
-        if (campaignStatus == null || !campaignStatus.OwnCampaignExists)
+        if (campaignStateData == null)
```

Comment:
I have not a strong opinion on this, but maybe when `campaignStatus.OwnCampaignExists` is `false` we should simply ignore the rest and simply return a `NoExistCampaign`.

```suggestion
        if (campaignStatus == null || !campaignStatus.OwnCampaignExists)
```